<launch>

    <arg name="use_gazebo" default="true" />
    <arg name="env_name" default="training_env"/>

    <!-- Gazebo -->
    <group if="$(arg use_gazebo)">
        <include file="$(find drl_navigation)/launch/training_env.launch">
            <arg name="env_name" value="$(arg env_name)" />
        </include>
        <include file="$(find rosbot_description)/launch/rosbot_gazebo.launch"/>
        <param name="use_sim_time" value="true" />
    </group>
    
    <node pkg="gmapping" type="slam_gmapping" name="gmapping">
        <param name="base_frame" value="base_link" />
        <param name="odom_frame" value="odom" />
        <param name="delta" value="0.05" />
        <param name="xmin" value="-5" />
        <param name="ymin" value="-5" />
        <param name="xmax" value="5" />
        <param name="ymax" value="5" />
        <param name="maxUrange" value="15" />
        <param name="map_update_interval" value="1" />
        <param name="linearUpdate" value="0.1" />
        <param name="angularUpdate" value="0.2" />
        <param name="temporalUpdate" value="0.1" />
        <param name="particles" value="100" />
        <param name="srr" value="0.4" />
        <param name="srt" value="0.5" />
        <param name="str" value="0.4" />
        <param name="stt" value="0.5" />
        <param name="iterations" value="6" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find drl_navigation)/rviz/slam_config.rviz" />

</launch>